src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs

```diff
@@ -170,10 +179,32 @@ public JArray GetAccounts()
         [JsonRpcMethod("eth_sign")]
         private string Sign(string address, string message)
         {
-            // TODO: implement message signing
-            //var addressUint160 = address.HexToUInt160();
-            return Web3DataFormatUtils.Web3Data("".HexToBytes());
-            //throw new ApplicationException("Not implemented yet");
+            if (address is null)
+            {
+                throw new ArgumentException("address should not be null");
+            }
+
+            byte[]? messageBytes = message.HexToBytes();
+
+            if (_privateWallet.IsLocked())
+            {
+                throw new Exception("wallet is locked");
+            }
+
+            var addressUInt160 = (address!).HexToUInt160();
+            if (addressUInt160.Equals(_privateWallet.EcdsaKeyPair.PublicKey.GetAddress()))
+            {
+                throw new Exception("address is invalid");
+            }
+
+            var keyPair = _privateWallet.EcdsaKeyPair;
+
+            Logger.LogInformation($"Keys: {keyPair.PublicKey.GetAddress().ToHex()}");
+
+            var messageHashBytes = messageBytes.KeccakBytes();
+            var signed = crypto.SignHashed(messageHashBytes, keyPair.PrivateKey.Encode());
```

Comment:
As I said,  don't just sign received message,  add some constant prefix to the message before signing,  for example sign smth like "LACHAIN" + messageLength + message